ui/src/views/Quickstart/index.jsx

```diff
@@ -204,36 +279,19 @@ export default class QuickStart extends Component {
   };
 
   handleReset = () => {
-    this.setState(initialState);
+    const resetState = {
+      ...this.initialState,
+      condition: getMatchCondition(this.initialState.events),
+    };
+
+    this.setState({
+      ...resetState,
+      editorValue: getNewYaml(resetState),
```

Comment:
I have mixed feelings about the name `getNewYaml`. Let's find a more descriptive name.

Reply:
How about `formatYml` or `generateYml` ?

Reply:
What is that we are generating/formatting with `formatYml` and `generateYml`? For example, if a function is responsible in returning the yml of a task definition, then I would probably name it something like `getTaskDefinition`. Let me know what you think.

Reply:
I was thinking about it in the other way, basically we have initialYml value defined as json
[Line 30](https://github.com/taskcluster/taskcluster/pull/199/files/66a5d75339efcf46929e1410f3f7cf8137abb546#diff-ab453666b30fa170ad0870e42fc3b8deR30)

And the json is getting formatted into YML, in `getNewYml` function, based on this thinking, I was suggesting, `formatYml` or `generateYml`.

Reply:
I think we can also rename `initialYml` to something else. The only reason we have `initialState` is because state means something to a React component. `initialYml` doesn't say much about what it is.

Reply:
Ok. So, function name will be `getTaskDefinition`
How about changing `initialYml` to just `yml` ?

Reply:
I don't think `yml` says a lot. What is inside that yml?

Reply:
It has json structure of the yml which is formatted into a real yml and its rendered in the editor.
https://github.com/taskcluster/taskcluster/pull/199/files#diff-ab453666b30fa170ad0870e42fc3b8deR30

Reply:
What does that json structure represent?

Reply:
It represents the data which will be converted to yml script and it will be used while running auto scripts to CI/CD

Reply:
> It represents the data which will be converted to yml script

What's inside that data? Is it a task definition? I think once you are able to answer this question, you'll be in a better measure of picking the right variable name :)

Reply:
Yes. One word for whatever I was trying to explain is `task defintion`.
I think we can name the variable `ymlTaskDefinition` or simply `taskDefinition`

Reply:
@helfi92 Does these variable names makes sense? I can update the PR with changed variable and function name once confirmation